src/main/java/seedu/address/logic/commands/AddNoteCommand.java

```diff
@@ -0,0 +1,110 @@
+    public static final String MESSAGE_DELETE_NOTE_SUCCESS = "Removed note from Person: %1$s";
+    private final IdentityCardNumberMatchesPredicate ic;
+    private final Note note;
+    private final boolean isReplace;
+
+    /**
+     * @param ic of the person in the filtered person list to edit the note
+     * @param note of the person to be updated to
+     */
+    public AddNoteCommand(IdentityCardNumberMatchesPredicate ic, Note note, boolean isReplace) {
+        requireAllNonNull(ic, note);
+
+        this.ic = ic;
+        this.note = note;
+        this.isReplace = isReplace;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        model.updateFilteredPersonList(ic);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (lastShownList.isEmpty()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON);
+        }
+
+        Person personToEdit = lastShownList.get(0);
+        if (isReplace) {
+            Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                    personToEdit.getIdentityCardNumber(), personToEdit.getAge(), personToEdit.getSex(),
+                    personToEdit.getAddress(), note, personToEdit.getTags());
+
+            model.setPerson(personToEdit, editedPerson);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+            return new CommandResult(generateSuccessMessage(editedPerson));
+        } else {
+            Note updatedNote = personToEdit.getNote().append("\n" + note.toString());
+            Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                    personToEdit.getIdentityCardNumber(), personToEdit.getAge(), personToEdit.getSex(),
+                    personToEdit.getAddress(), updatedNote, personToEdit.getTags());
+
+            model.setPerson(personToEdit, editedPerson);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+            return new CommandResult(generateSuccessMessage(editedPerson));
+        }
+    }
+
+    /**
+     * Generates a command execution success message based on whether the remark is added to or removed from
+     * {@code personToEdit}.
+     */
+    private String generateSuccessMessage(Person personToEdit) {
+        String message = !note.value.isEmpty() ? MESSAGE_ADD_NOTE_SUCCESS : MESSAGE_DELETE_NOTE_SUCCESS;
+        return String.format(message, personToEdit);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof AddNoteCommand)) {
+            return false;
+        }
+
+        AddNoteCommand e = (AddNoteCommand) other;
+        return ic.equals(e.ic)
+                && note.equals(e.note);
```

Comment:
Do we need to check `isReplace` is the same too?

Reply:
Oh yes sorry I think I missed it out! Thanks